$OpenBSD: patch-libavcodec_aacps_c,v 1.1 2013/08/19 14:38:29 brad Exp $

Correct opdipd code to match spec. This fixes out of array reads.

--- libavcodec/aacps.c.orig	Wed Jun 12 19:06:09 2013
+++ libavcodec/aacps.c	Wed Jun 12 19:07:00 2013
@@ -823,7 +823,8 @@ static void stereo_processing(PSContext *ps, float (*l
             h12 = H_LUT[iid_mapped[e][b] + 7 + 23 * ps->iid_quant][icc_mapped[e][b]][1];
             h21 = H_LUT[iid_mapped[e][b] + 7 + 23 * ps->iid_quant][icc_mapped[e][b]][2];
             h22 = H_LUT[iid_mapped[e][b] + 7 + 23 * ps->iid_quant][icc_mapped[e][b]][3];
-            if (!PS_BASELINE && ps->enable_ipdopd && b < ps->nr_ipdopd_par) {
+
+            if (!PS_BASELINE && ps->enable_ipdopd && 2*b <= NR_PAR_BANDS[is34]) {
                 //The spec say says to only run this smoother when enable_ipdopd
                 //is set but the reference decoder appears to run it constantly
                 float h11i, h12i, h21i, h22i;
